CHANGELOG.md

```diff
@@ -1,5 +1,162 @@
+
+### Field
+
+- **Breaking change:** Support implicit `Field.Label`.
+  If `Field.Label` encloses Switch/Checkbox/Radio, the `htmlFor`/`id` attributes are no longer explicitly set to associate them.
+  (#2036) by @mj12albert
+- Enable custom validation based on other form values (#1941) by @mj12albert
+- Fix `onValueChange` `value` type (#2112) by @atomiks
+- Fix `Field.Label` focusing trigger (#2118) by @atomiks
+- Fix slider field label (#2154) by @mj12albert
+
+### Fieldset
+
+- Refactor to `useRenderElement` (#2053) by @mj12albert
+
+### Form
+
+- Enable custom validation based on other form values (#1941) by @mj12albert
+
+### Input
+
+- Fix `onValueChange` `value` type (#2112) by @atomiks
+
+### Menu
+
+- **Breaking change:** Add `SubmenuRoot` part.
+  Nested menus should be defined with `Menu.SubmenuRoot` instead of `Menu.Root` to to avoid ambiguity.
+  (#2042) by @atomiks
+- Unset `role` from Trigger (#2047) by @atomiks
+- Emit `close` event on `cancel-open` (#2067) by @atomiks
+- Fix close toggle when rendering non-native button (#2071) by @atomiks
+- Add `highlighted` to item `State` (#2079) by @atomiks
+- Remove highlighted effect (#2162) by @atomiks
+- Cut out internal backdrop to allow interacting with triggers (#2141) by @michaldudak
+- Fix active index sync on hover (#2163) by @atomiks
+
+### Menubar
+
+- Fix `closeOnClick: false` not working in nested menus (#2094) by @michaldudak
+
+### Navigation Menu
+
+- Fix typos (#2139) by @atomiks
+- Handle layout resize while open (#2070) by @atomiks
+- Fix positioner height when opening menu using the keyboard arrows (#2060) by @juliomerisio
+
+### Number Field
+
+- Ensure `onValueChange` is called with already-formatted parsed value (#1905) by @atomiks
+- Fix revalidation on change (#2174) by @atomiks
+
+### Popover
+
+- Fix close toggle when rendering non-native button (#2071) by @atomiks
+- Cut out internal backdrop to allow interacting with triggers (#2141) by @michaldudak
+
+### Radio Group
+
+- **Breaking change:** Support implicit `Field.Label`.
+  If `Field.Label` encloses Radio, the `htmlFor`/`id` attributes are no longer explicitly set to associate them.
+  (#2036) by @mj12albert
+- Refactor to `useRenderElement` (#2053) by @mj12albert
+
+### Scroll Area
+
+- Ignore `data-scrolling` during programmatic scroll (#1908) by @atomiks
+- Handle touch momentum scrolling for programmatic scroll check (#2111) by @atomiks
+
+### Select
+
+- **Breaking change:** Print raw value in `Select.Value`.
+  `<Select.Value>` now prints the raw value by default unless an `items` prop is specified on `Select.Root`.
+  (#2087) by @atomiks
+- Performance: avoid re-renders (#1961) by @romgrk
+- Fix close toggle when rendering non-native button (#2071) by @atomiks
+- Fix re-render warning in `Positioner` on fallback (#2089) by @atomiks
+- Fix `Field.Label` focusing trigger (#2118) by @atomiks
+- Fix position flicker when no default value is set (#2110) by @atomiks
```

Comment:
A bunch of these were made as the result of regressions post-`beta.0` release and don't need to be mentioned:

 - Fix re-render warning in `Positioner` on fallback
 - Fix position flicker when no default value is set
 - Fix inverted flag in item mouseleave
 - Fix revalidation on change